Bottom: 6c1e4cf53330ee6255062b2aa3a77c802dd15072
Top:    f8ed0056a30733328719d514d1c06de875638561
Author: Johannes Rudolph <johannes_rudolph@gmx.de>
Date:   2010-05-31 10:23:19 +0200

fix method dispatch (changed varargs calling convention in Scala 2.8.0)


---

diff --git a/objectformatter/src/main/scala/net/virtualvoid/string/ESParser.scala b/objectformatter/src/main/scala/net/virtualvoid/string/ESParser.scala
index d7f4d09..2b9db8b 100644
--- a/objectformatter/src/main/scala/net/virtualvoid/string/ESParser.scala
+++ b/objectformatter/src/main/scala/net/virtualvoid/string/ESParser.scala
@@ -63,7 +63,7 @@ object AST{
     import java.lang.NoSuchMethodException
     def findMethod(c:Class[_],name:String):Option[Method] =
      try {
-       val res = c.getMethod(name,null)
+       val res = c.getMethod(name)
        res.setAccessible(true)
        Some(res)
      }catch{
@@ -83,7 +83,7 @@ object AST{
     def returnType(callingCl:Class[_]):Class[_] = method(callingCl).getReturnType
     def genericReturnType(callingCl:Class[_]):java.lang.reflect.Type = method(callingCl).getGenericReturnType
     def capitalize(s:String):String = s.substring(0,1).toUpperCase + s.substring(1)
-    def eval(o:AnyRef) = method(o.getClass).invoke(o,null)
+    def eval(o:AnyRef) = method(o.getClass).invoke(o)
   }
   case object ThisExp extends Exp(""){
     override def eval(o:AnyRef) = o
